### PR TITLE
Use msbIdx for direct lookup, and then additional check if prev is la…

### PR DESCRIPTION
**Pull request by baldersheim**
…rge enough.

@vekterli or @jonmv PR
This supeceeds #31993

### CODE REVIEW

**Review by jonmv — DISMISSED** [review]

File: vespalib/src/tests/stllike/hashtable_test.cpp [hunk 1]
```diff
@@ -69,9 +67,24 @@ TEST("require that hashtable<int> can be copied") {
...
+        EXPECT_GREATER_EQUAL(prime, num);
+        printf("%lu <= %lu\n", num, prime);
+    }
+    for (size_t i=0; i < 32; i++) {
+        size_t num = (1ul << i) - 1;
+        size_t prime = hashtable_base::getModuloStl(num);
+        EXPECT_GREATER_EQUAL(prime, num);
+        printf("%lu <= %lu\n", num, prime);
+    }
+}
```
Comment (jonmv) [inline vespalib/src/tests/stllike/hashtable_test.cpp:83]: Perhaps also test all the primes, and their ++1?

Comment (baldersheim) [reply vespalib/src/tests/stllike/hashtable_test.cpp:85]: Done

File: vespalib/src/tests/stllike/hashtable_test.cpp [hunk 2]
```diff
@@ -69,9 +67,26 @@ TEST("require that hashtable<int> can be copied") {
     EXPECT_EQUAL(42, *table2.find(42));
 }
 
+TEST("require that getModuloStl always return a larger number in 32 bit integer range") {
+    for (size_t i=0; i < 32; i++) {
+        size_t num = 1ul << i;
+        size_t prime = hashtable_base::getModuloStl(num);
+        EXPECT_GREATER_EQUAL(prime, num);
+        EXPECT_EQUAL(prime, hashtable_base::getModuloStl(prime));
+        EXPECT_GREATER(hashtable_base::getModuloStl(prime+1), prime + 1);
```
Comment (jonmv) [inline vespalib/src/tests/stllike/hashtable_test.cpp:76]: This can only work because `0 < 7` (for `i = 31`), but ok 😁

**Review by vekterli — COMMENTED** [review]
:+1: just to verify, I ran a brute force check between the old and new code for all values in `[0, UINT32_MAX]` and the outputs were identical for the same input size

**Pull request merged**